src/main/java/avd/inf/jdm/rentmycar/controller/UserController.java

```diff
@@ -57,7 +57,7 @@ public ResponseEntity<Object> getAllUsers() {
     public ResponseEntity<Object> createUser(@Valid @RequestBody UserDto userDto) {
         User newUser;
         try {
-            User maybeUser = userService.getUserByEmail(userDto.getEmail());
+            Optional<User> maybeUser = userService.getUserByEmail(userDto.getEmail());
```

Comment:
And here as well. An `Optional` is never `null` as far as I know, is it is just empty if the user can not be found.

Reply:
good one! fixed that